manager-bundle/src/HttpKernel/ContaoKernel.php

```diff
@@ -193,8 +194,49 @@ public function setManagerConfig(ManagerConfig $managerConfig): void
      */
     public function registerContainerConfiguration(LoaderInterface $loader): void
     {
-        if ($configFile = $this->getConfigFile('parameters.yml')) {
-            $loader->load($configFile);
+        if ($parametersFile = $this->getConfigFile('parameters.yml')) {
+            $loader->load($parametersFile);
+
+            // Set the .env variables from the parameters.yml file (backwards compatibility)
+            $loader->load(
+                static function (ContainerBuilder $container): void {
+                    if (!isset($_SERVER['APP_SECRET']) && $container->hasParameter('secret')) {
+                        $container->setParameter('env(APP_SECRET)', $container->getParameter('secret'));
+                    }
+
+                    if (!isset($_SERVER['DATABASE_URL']) && $container->hasParameter('database_host')) {
+                        $container->setParameter(
+                            'env(DATABASE_URL)',
+                            sprintf(
+                                'mysql://%s:%s@%s:%s/%s',
+                                rawurlencode($container->getParameter('database_user')),
+                                rawurlencode($container->getParameter('database_password')),
+                                rawurlencode($container->getParameter('database_host')),
+                                (int) $container->getParameter('database_port'),
+                                rawurlencode($container->getParameter('database_name'))
+                            )
+                        );
+                    }
+
+                    if (!isset($_SERVER['MAILER_URL']) && $container->hasParameter('mailer_transport')) {
+                        if ('sendmail' === $container->getParameter('mailer_transport')) {
+                            $container->setParameter('env(MAILER_URL)', 'sendmail://localhost');
+                        } elseif ('smtp' === $container->getParameter('mailer_transport')) {
+                            $container->setParameter(
+                                'env(MAILER_URL)',
+                                sprintf(
```

Comment:
I think we should have more checks here whether the fields are empty. I might connect to a local SMTP server without username and password.

Reply:
Changed in 3539567db526978673a508199bef2228cddc1e62.